src/components/tree-views-container.jsx

```diff
@@ -315,7 +315,7 @@ const TreeViewsContainer = () => {
     const mergeCurrentAndUploading = useCallback(
         (current) => {
             let elementsToMerge = Object.values(uploadingElements).filter(
-                (e) => e.directory === selectedDirectoryRef.current.elementUuid && current[e.elementName] === undefined
+                (e) => e.directory === selectedDirectoryRef.current?.elementUuid && current[e.elementName] === undefined
```

Comment:
Not for this scope I think and why this change ?

Reply:
I can't seem to replicate the situation where the **selectedDirectoryRef.current** is null, which caused the app to crash in this particular scenario.
However, since **selectedDirectoryRef.current** is instantiated with the **selectedDirectory**, which may be null, I believe we should maintain this check.